ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageContainerMetricsCollector.java

```diff
@@ -0,0 +1,96 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A class that aggregates container metrics. We do _NOT_ want to emit per-container metrics because then the number
+ * of metrics increases proportionally overwhelming telemetry. This was attempted and failed.
+ *
+ * A daemon will run at regular intervals emitting aggregate metrics in a controlled and predictable manner.
+ * This is a singleton class to avoid multiple collector threads.
+ */
+public class AzureStorageContainerMetricsCollector {
+  private final AzureMetrics metrics;
+  private final ConcurrentHashMap<Long, AzureStorageContainerMetrics> metricMap;
+  private final ScheduledExecutorService executor;
+  private static AzureStorageContainerMetricsCollector instance;
+  private final VerifiableProperties properties;
+
+  public static final Logger logger = LoggerFactory.getLogger(AzureStorageContainerMetricsCollector.class);
+
+  private AzureStorageContainerMetricsCollector(MetricRegistry metrics, VerifiableProperties properties) {
+    metricMap = new ConcurrentHashMap<>();
+    this.metrics = new AzureMetrics(metrics);
+    this.properties = properties;
+    executor = Utils.newScheduler(1, "azure_storage_container_metrics_collector_", true);
+    executor.scheduleWithFixedDelay(getCollector(), 0, 2, TimeUnit.MINUTES);
+    logger.info("Started AzureStorageContainerMetricsCollector");
+  }
+
+  private Runnable getCollector() {
+    return () -> {
+      Long totalDrift = metricMap.values().stream()
+          .map(container -> container.getDrift())
+          .reduce(0L, Long::sum);
+      this.metrics.azureContainerDriftBytesCount.inc(totalDrift);
+    };
+  }
+
+  /**
+   * Thread-safe singleton initializer
+   * @param metrics
+   * @return collector instance
+   */
+  public static synchronized AzureStorageContainerMetricsCollector getInstance(MetricRegistry metrics,
+      VerifiableProperties properties) {
+    if (instance == null) {
+      instance = new AzureStorageContainerMetricsCollector(metrics, properties);
+    }
+    return instance;
+  }
+
+  public void addContainer(Long id) {
+    metricMap.put(id, new AzureStorageContainerMetrics(id));
+  }
+
+  public void removeContainer(Long id) {
+    metricMap.remove(id);
+  }
+
+  /**
+   * Sets the drift of azure-container from ambry-partition.
+   * We use a compare-set to guard against accidental multithreaded errors, although two threads will most likely
+   * not be responsible for a single partition in VCR. A single thread handles all replicas of a partition.
+   * However, we want to avoid any races between reader and writers.
+   * Use min() as bootstrapping replicas can give a wrong picture and indicate a large drift even though the partition
+   * is fully backed up in Azure.
+   * @param id
+   * @param drift
+   */
+  public void setContainerDrift(long id, long drift) {
```

Comment:
Drift is probably not a good variable name. I suppose lag would be better given the context here.

Reply:
Changed to `lag`